### PR TITLE
[AIX] use LIBPATH on AIX instead of LD_LIBRARY_PATH

### PR DESCRIPTION
**Pull request by chenzheng1030**
LD_LIBRARY_PATH will become invalid when LIBPATH is also set on AIX.

See below example on AIX:
```
$ldd a.out
a.out needs:
	 /usr/lib/libc.a(shr.o)
Cannot find libtest.a
	 /unix
	 /usr/lib/libcrypt.a(shr.o)

$./a.out
Could not load program ./a.out:
	Dependent module libtest.a could not be loaded.
Could not load module libtest.a.
System error: No such file or directory

$export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/tmp
$./a.out ; echo $?
10

$export LIBPATH=./
$./a.out ; echo $?  >>>>>> Now LD_LIBRARY_PATH is not used by system loader
Could not load program ./a.out:
	Dependent module libtest.a could not be loaded.
Could not load module libtest.a.
System error: No such file or directory
```

This breaks many AIX LIT cases on our downstream buildbots which sets LIBPATH.

### CODE REVIEW

File: llvm/utils/lit/lit/llvm/config.py [hunk 1]
```diff
@@ -588,7 +588,10 @@ def use_clang(
             if getattr(self.config, pp, None)
         ]
 
-        self.with_environment("LD_LIBRARY_PATH", lib_paths, append_path=True)
+        if sys.platform.startswith("aix"):
```
Comment (RoboTux) [inline llvm/utils/lit/lit/llvm/config.py:591]: Nit: I see more cases of platform.system() == "AIX" in the LIT code, could we use that instead?

**Review by RoboTux — COMMENTED** [review]
LGTM otherwise

File: llvm/utils/lit/lit/llvm/config.py (hunk 1 above)
Comment (anhtuyenibm) [reply llvm/utils/lit/lit/llvm/config.py:591]: I agreed with RoboTux @RoboTux, can we try something like line 106
```
        elif platform.system() == "AIX":
            features.add("system-aix") 
```

**Review by anhtuyenibm — APPROVED** [review]
LGTM with a minor suggestion.

**Review by daltenty — APPROVED** [review]
LGTM as well with the minor adjustment proposed

File: llvm/utils/lit/lit/llvm/config.py (hunk 1 above)
Comment (daltenty) [reply llvm/utils/lit/lit/llvm/config.py:591]:
```suggestion
        if platform.system() == "AIX":
```

**Pull request merged**